### PR TITLE
Update Journal window to Gtk 2.24, drop Gnome and use GtkBuilder

### PR DESCRIPTION
**Pull request by oskarb**
This updates the Journal dialog to "modern" Gtk2 and GtkBuilder.

  * Regenerates the journal.glade file to target Gtk 2.24 and drops the Gnome components.
  * Converts the journal.glade file to GtkBuilder format.

### CODE REVIEW

File: src/design.txt
Comment (markuspg) [inline src/design.txt]: I'm sorry for the nitpick, but _WebKitGTK_ please.

```suggestion
   of project activity as html, inside the WebKitGTK html
```

**Comment by markuspg** [comment]
I'm sorry for the belated review. I wouldn't block this pull request, but I wonder if it makes sense to wait with the migration from GtHTML to WebKitGTK until we move from GTK 2 to GTK 3. By this we could save a few intermediary steps by migrating to the most up-to-date WebKitGTK version supporting GTK 3. Additionally I had the issue on my tries that I was not able to replicate the functionality of GtkHTML with early WebKitGTK versions. Newer WebKitGTK versions are a clear superset of functionality of GtkHTML.

File: src/util.h
Comment (markuspg) [inline src/util.h]: As written in #11 I would see this as a candidate for a separate pull request making a minimal GtkBuilder spike. Pulling this out helps us integrating it faster and makes this pull request focussed on the GtkHTML to WebKitGTK migration.

**Comment by oskarb** [comment]
> if it makes sense to wait with the migration from GtHTML to WebKitGTK

Well, this whole branch was to experiment, but it could make a lot of sense to at least take WebKitGTK out of this PR, to allow the other parts to be merged more easily if you agree on the direction taken.

As for going directly to webkit2gtk at a later point, I'm not sure. I tried to look at the webkit1gtk and webkit2gtk APIs and my impression is that a lot is the same or similar, especially considering our limited use. So it might be worth it to land the webkit1gtk version.


> Additionally I had the issue on my tries that I was not able to replicate the functionality of GtkHTML with early WebKitGTK versions. 

Can you elaborate a bit on that? Currently in this PR some signal callbacks are commented out, but I haven't been able to figure out anything that's not working. Except the GnoTime logo doesn't load, but at least for me it doesn't load in gtkhtml either and I would prefer to just get rid of it, or possibly embed it as a data url. Currently, if the report is saved to HTML and viewed externally, the logo won't work either.


--

Anyway, I reordered the commits so that the webkit work comes last. If you agree on the direction taken here (GtkBuilder), after #29 and #34 are merged, I will rebase, test a bit more, and submit for review without webkit. I am so far planning/proposing to keep the naive use of `gtk_builder_connect_signals_full` and `connect_signals_cb` at this time, and explore using GModule (or at the very least implement reusable helpers for connect_signals_cb) in a separate PR.

**Comment by oskarb** [comment]
@markuspg I've rebased and taken the webkit work out of this PR. I now consider this complete. Please review.

**Comment by markuspg** [comment]
I wonder if it would be cleaner to rename the file upon conversion from _Glade_ to _GtkBuilder_ from `journal.glade` to `journal.ui` to properly reflect the new storage format. Furthermore I would suggest to move the file from the `glade` subdirectory to a new `ui` subdirectory.

**Comment by markuspg** [comment]
How are you editing this in _Glade_? I'm using `Glade 2.12.2`  (on `Fedora 37`) and I get a "document not well formed!" error - but unfortunately no more elaborate error message. _Glade_ still starts, but there is no open project then.

**Comment by oskarb** [comment]
In short, you want to use glade 3.8.x (but not later). This is available as
a package on Ubuntu 14.

I think it's strange that recent Fedora carries glade 2.12 and the latest
3.x. It's true that 2.12 is the last in the glade 2.x series, but glade 3.8
is the really last version to support GTK 2, including the last version of
GTK 2 (2.24), and support for saving in GtkBuilder format.

The glade project itself notes that 3.8 is intended specifically as the
migration path (https://blogs.gnome.org/tvb/2011/01/15/the-glade-dl/).
Basically, if you have got to GTK 2.24 with no deprecation warnings and
using gtkbuilder format in glade 3.8, it should then already be usable in
glade 3.10 and gtk 3.

**Comment by oskarb** [comment]
> I wonder if it would be cleaner to rename the file upon conversion from _Glade_ to _GtkBuilder_ from `journal.glade` to `journal.ui` to properly reflect the new storage format. Furthermore I would suggest to move the file from the `glade` subdirectory to a new `ui` subdirectory.

Agreed.

**Review by markuspg — APPROVED** [review]

File: src/util.h
Comment (oskarb) [reply src/util.h]: Unsure if I saw these suggestions before - memory foggy right now. But I did see the comments in #11 anyway and separated some changes to a separate PR and the remainder to several commits that I had intended to be merged separately inside this PR.

File: src/design.txt
Comment (oskarb) [reply src/design.txt]: Applied quickly as a fixup in draft #36 - I will squash that later when I rebase it.

**Pull request merged**
